haystack/nodes/retriever/dense.py

```diff
@@ -1877,6 +1877,8 @@ def _preprocess_documents(self, docs: List[Document]) -> List[Document]:
     def _infer_model_format(model_name_or_path: str, use_auth_token: Optional[Union[str, bool]]) -> str:
         if any(m in model_name_or_path for m in ["ada", "babbage", "davinci", "curie"]):
             return "openai"
+        if any(m in model_name_or_path for m in ["small", "medium", "large"]):
```

Comment:
Here, I would check if `model_name_or_path` is strictly equal to `"small"`, `"medium"` or `"large"`, as these words are likely to appear in non-cohere models as well, for example [sentence-transformers/all-roberta-large-v1](https://huggingface.co/sentence-transformers/all-roberta-large-v1).

Reply:
Great catch! Thank you